yt_dlp/extractor/bilibili.py

```diff
@@ -248,6 +244,11 @@ def _real_extract(self, url):
                         'quality': -3,
                     })
 
+            for a_format in formats:
+                a_format.setdefault('http_headers', {}).update({
+                    'Referer': url,
+                })
+
```

Comment:
u can just put the http_headers in the info_dict directly

Reply:
I just fix this ❤️